core/src/autogluon/core/utils/deprecated.py

```diff
@@ -0,0 +1,19 @@
+import logging
+import warnings
+import functools
+
+
+logger = logging.getLogger(__name__)
+
+
+def deprecated(message="", *args, **kwargs):
+    def deprecated_decorator(func):
+        warning_message = f"Calling deprecated function {func.__name__}. This function is deprecated " \
```

Comment:
Nit: Should we use parenthesis for multi-line strings? For example (or we can also move `+ message` to the `logger` call
```
default_warning_ = (
    f"Calling deprecated function {func.__name__}. This function is deprecated "
    f"and will be removed in a future release. " 
    + message
)
```